param/__init__.py

```diff
@@ -102,6 +103,27 @@ def named_objs(objlist):
     return objs
 
 
+def param_union(*parameterizeds, **kwargs):
+    """
+    Given a set of Parameterized objects, returns a dictionary
+    with the union of all param name,value pairs across them.
+    If warn is True (default), warns if the same parameter has
+    been given multiple values; otherwise uses the last value
+    """
+    warn = kwargs.pop('warn', True)
+    if len(kwargs):
+        raise TypeError(
+            "param_union() got an unexpected keyword argument '{}'".format(
+                kwargs.popitem()[0]))
+    d = dict()
+    for o in parameterizeds:
+        for k, p in o.param.params().items():
+            if k != 'name':
+                if k in d and warn:
+                    warnings.warn("overwriting parameter {}".format(k))
```

Comment:
My understanding is that this would show a warning only one time ever (i.e. only the first time any overwrite happens). Unless you definitely know otherwise, could you check?

Reply:
By the way, param has its own messaging/warning system. While I don't particularly want to recommend it because it's annoying in many ways, it avoids problems like the above (and using it would be consistent with the rest of param, in general). So maybe `Parameterized(name="param_union").warning("overwriting parameter %s",k)` (untested).

Reply:
It happens once (and only once) for each parameter the overwrite happens with. So yes, switching to param's warning system makes a lot of sense (I just took precedent from ``version.py``). I'll also update the test for posterity. Is there any way to avoid the 'root' part of the message using your proposed method?
```
WARNING:root:param_union: overwriting parameter a
WARNING:root:param_union: overwriting parameter b
```
I don't know what purpose 'root' serves in general. Please let me know what you think :)

Reply:
> It happens once (and only once) for each parameter the overwrite happens with

Thanks for checking and for staying engaged through these seemingly trivial details!

> I just took precedent from version.py

Sorry to confuse you! That file is for versioning rather than being related to parameters; I'm not familiar with its internals.

> Is there any way to avoid the 'root' part of the message using your proposed method
> I don't know what purpose 'root' serves in general

I'm sure there's a way to avoid the 'root' part of the message, but it wouldn't be specific to this work on `param_union()`. I actually wasn't (consciously) aware that 'root' is shown, nor do I know what purpose it serves. param's logging system uses Python's logging module (see https://github.com/ioam/param/blob/master/param/parameterized.py#L32); I think everyone would be happy to see it simplified and/or clarified, but nobody's had time to work on it.

Reply:
'root' doesn't serve any important purpose; it arises simply because the logging system is designed to log warnings and errors for particular Parameterized classes and instances, where "root" would be replaced with the actual class or instance raising the issue.  Here, it's the "root" or base class that's raising it, as a degenerate case, and we should probably just remove the object name altogether in that case.  That would be a backwards-incompatible change for anyone who searches or filters the log messages, but it's something that could be done in our upcoming shift to Param 2.0.

Reply:
Oops; you're right; I wasn't looking closely enough and I was talking about a completely different problem, i.e. that to get a message not tied to an existing Parameterized, one has to use a dummy instance of Parameterized(), which then shows up in the messages inappropriately.  "root" must be from the logger, as you say, in which case should we be using that bit for the Parameterized instance or class information (collapsing "root" and "param_union" above)?

Reply:
Or maybe we should just declare that the root logger has the name "param", to give people an idea where these messages come from?

Reply:
Any decisions, gentleman?

Also, anyone feel like taking a look on my comment in #159?

Reply:
I've moved this discussion to https://github.com/ioam/param/issues/267 so that I could merge this PR.